### PR TITLE
Feature/235 get flow partitionings

### LINKED ISSUE #235: GET /flows/{flowId}/partitonings
## Feature
Implement `GET /flows/{flowId}/partitonings` V2 REST API endpoint on Server.

Returns all the partitioning belonging to the flow with PAGINATION

## Proposed Solution
```scala
protected val getFlowPartitioningsEndpointV2
    : PublicEndpoint[(Long, Option[Int], Option[Long]), ErrorResponse, PaginatedResponse[
      PartitioningWithIdDTO
    ], Any] = {
    apiV2.get
      .in(V2Paths.Flows / path[Long]("flowId") / V2Paths.Partitionings)
      .in(query[Option[Int]]("limit"))
      .in(query[Option[Long]]("offset"))
      .out(statusCode(StatusCode.Ok))
      .out(jsonBody[PaginatedResponse[PartitioningWithIdDTO]])
  }
```

### PR DESCRIPTION
**Pull request by salamonpavel**
Introduces GET endpoint for retrieval of partitionings of a given flow
Closes #235

### CODE REVIEW

**Comment by salamonpavel** [comment]
Release notes:

- Introduces GET endpoint for retrieval of partitionings of a given flow

File: database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+CREATE OR REPLACE FUNCTION flows.get_flow_partitionings(
+    IN i_flow_id               BIGINT,
+    IN i_limit                 INT DEFAULT 5,
+    IN i_offset                BIGINT DEFAULT 0,
+    OUT status                 INTEGER,
+    OUT status_text            TEXT,
+    OUT id                     BIGINT,
+    OUT partitioning           JSONB,
+    OUT author                 TEXT,
+    OUT has_more               BOOLEAN
```
Comment (lsulak) [inline database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql:26]: do we also want to care whether a given partitioning was `primary` for the given flow?

**Review by lsulak — COMMENTED** [review]
Review finished, apart from 1 q from me and 1 note from David it looks good

File: database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql (hunk 1 above)
Comment (benedeki) [reply database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql:26]: Little value now, and adds a table to the JOIN.
IMHO not worth it.
Btw, we could add the main flow id to partitioning table (for easier search and one less index).

File: database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            _has_more AS has_more
+        FROM
+            runs.partitionings P
+        WHERE
+            P.id_partitioning IN (SELECT LP.id_partitioning FROM limited_partitionings LP)
+        ORDER BY
+            P.created_at DESC,
+            P.id_partitioning;
+
+    IF NOT FOUND THEN
```
[the remark this answers is not in the document]
Comment (salamonpavel) [reply database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql:101]: ok changed

File: database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            P.partitioning,
+            P.created_by AS author,
+            _has_more AS has_more
+        FROM
+            runs.partitionings P
+        WHERE
+            P.id_partitioning IN (SELECT LP.id_partitioning FROM limited_partitionings LP)
+        ORDER BY
+            P.created_at DESC,
+            P.id_partitioning;
```
[the remark this answers is not in the document]
Comment (salamonpavel) [reply database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql:99]: ok changed

File: database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            P.id_partitioning,
+            P.created_at DESC
+        LIMIT i_limit OFFSET i_offset;
+
+    IF NOT FOUND THEN
+        status := 12;
+        status_text := 'OK with no partitionings found';
+        RETURN NEXT;
+    END IF;
+END;
```
Comment (benedeki) [inline database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql:100]:
```suggestion
```

File: database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        ORDER BY
+            P.id_partitioning,
+            P.created_at DESC
+        LIMIT i_limit OFFSET i_offset;
+
+    IF NOT FOUND THEN
+        status := 12;
+        status_text := 'OK with no partitionings found';
+        RETURN NEXT;
+    END IF;
```
Comment (benedeki) [inline database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql:99]:
```suggestion
```

File: database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+--      status_text             - Status text
+--      id                      - ID of retrieved partitioning
+--      partitioning            - Partitioning value
+--      author                  - Author of the partitioning
+--      has_more                - Flag indicating if there are more partitionings available
+--
+-- Status codes:
+--      11 - OK
+--      12 - OK with no partitionings found
+--      41 - Flow not found
```
Comment (benedeki) [inline database/src/main/postgres/flows/V1.9.8__get_flow_partitionings.sql:51]:
```suggestion
--      41 - Flow not found
```

**Pull request merged**